src/Configuration/Config.Abstractions/src/IConfigurationStreamLoader.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.IO;
+
+namespace Microsoft.Extensions.Configuration
+{
+    /// <summary>
+    /// Loads configuration key/values from a stream into a provider.
+    /// </summary>
+    public interface IConfigurationStreamLoader
```

Comment:
Why do we need this to be public?

Reply:
The base ConfigurationStreamLoader has a Loader property which is what the Json/Xml/Ini providers set to the appropriate type in the builder AddXyzStream() APIs

Reply:
I don’t think this interface should be public. Let’s make it internal and make the AddStream taking a loader internal as well. We can leave the Read helpers on the concrete types public

Reply:
So are you saying do internals visible to all the specific provider packages?

Reply:
The only drawback I see with that is other providers that aren't are own won't be able to expose a stream overload as easily but I guess they can just copy the code

Reply:
I think the code sharing here is minimal and not worth exposing a new public method. The user facing methods are Add{format}Stream and that's it. `AddStream(IConfigurationStreamLoader)` overload seems unnecessary

Reply:
Ok so if we don't want this concept in abstractions, rather than going internal, I can probably just make `StreamConfigurationProvider` abstract with `abstract void Load(Stream stream)` that won't require any internals visible

Reply:
Seems fine to me. I'd prefer that.